crates/import/src/rest.rs

```diff
@@ -0,0 +1,664 @@
+) -> anyhow::Result<Template> {
+    // Rest templates allow spaces in variables
+    // For example `{{ HOST}}` or `{{ HOST }}`
+    // These must be removed before putting it through the slumber
+    // template parser
+    let raw_template = template
+        .parts
+        .into_iter()
+        .map(|part| match part {
+            RestTemplatePart::Text(text) => text,
+            RestTemplatePart::Variable(var) => {
+                "{{".to_string() + var.as_str() + "}}"
+            }
+        })
+        .join("");
+
+    raw_template
+        .parse()
+        .map_err(|err| anyhow!("Failed to parse REST template! {err}"))
+}
+
+/// Convert a map of REST templates to Slumber templates (like headers or
+/// queries)
+/// Errors will be logged and skipped if an invalid template is passed in
+fn build_slumber_templates(
+    template_map: IndexMap<String, RestTemplate>,
+) -> IndexMap<String, Template> {
+    template_map
+        .into_iter()
+        .filter_map(|(k, v)| match try_build_slumber_template(v) {
+            Ok(t) => Some((k, t)),
+            Err(err) => {
+                error!("{err}");
+                None
+            }
+        })
+        .collect()
+}
+
+/// Convert REST Authentication to Slumber Authentication
+fn build_authentication(r_auth: RestAuthorization) -> Authentication {
+    match r_auth {
+        RestAuthorization::Bearer(bearer) => {
+            Authentication::Bearer(Template::raw(bearer))
+        }
+        RestAuthorization::Basic { username, password } => {
+            Authentication::Basic {
+                username: Template::raw(username),
+                password: password.map(|p| Template::raw(p)),
+            }
+        }
+    }
+}
+
+/// REST supports loading bodies from external files,
+/// this is connected to the request.
+/// In slumber, this needs to be converted into a request and
+/// a chain
+fn try_build_chain_from_load_body(
+    filepath: RestTemplate,
+    recipe_id: &str,
+    headers: &IndexMap<String, RestTemplate>,
+    variables: &RestVariables,
+) -> anyhow::Result<Chain> {
+    let full_id = format!("{recipe_id}_body");
+    let id: Identifier = Identifier::escape(&full_id);
+
+    let path = try_build_slumber_template(filepath)?;
+
+    let content_type =
+        guess_is_json(headers, variables).then(|| ContentType::Json);
+
+    Ok(Chain {
+        id: id.into(),
+        content_type,
+        trim: ChainOutputTrim::None,
+        source: ChainSource::File { path },
+        sensitive: false,
+        selector: None,
+        selector_mode: SelectorMode::Single,
+    })
+}
+
+/// Attempt to use headers to determine if the request is JSON
+fn guess_is_json(
+    r_headers: &IndexMap<String, RestTemplate>,
+    variables: &RestVariables,
+) -> bool {
+    for (name, value) in r_headers {
+        if name.to_lowercase() == CONTENT_TYPE.as_str() {
+            return value.render(variables) == APPLICATION_JSON.to_string();
+        }
+    }
+    false
```

Comment:
I think the functional form is more readable (once it's been formatted at least)
```suggestion
    r_headers.iter().any(|(name, value)| name.to_lowercase() == header::CONTENT_TYPE.as_str() && value.render(variables).as_str() == mime::APPLICATION_JSON)
```